src/core/mac/data_poll_handler.hpp

```diff
@@ -0,0 +1,271 @@
+    };
+
+    /**
+     * This class defines the callbacks used by the `DataPollHandler`.
+     *
+     */
+    class Callbacks : public InstanceLocator
+    {
+        friend class DataPollHandler;
+
+    private:
+        /**
+         * This constructor initializes the data poll handler object.
+         *
+         * @param[in]  aInstance   A reference to the OpenThread instance.
+         *
+         */
+        explicit Callbacks(Instance &aInstance);
+
+        /**
+         * This callback method requests a frame to be prepared for indirect transmission to a given sleepy child.
+         *
+         * @param[in] aFrame  A reference to a MAC frame where the new frame would be placed.
+         * @param[in] aChild  The child for which to prepare the frame.
+         *
+         * @retval OT_ERROR_NONE   Frame was prepared successfully
+         * @retval OT_ERROR_ABORT  Indirect transmission to child should be aborted (no frame for the child).
+         *
+         */
+        otError PrepareFrameForChild(Mac::Frame &aFrame, Child &aChild);
+
+        /**
+         * This callback method notifies the end of indirect frame transmission to a child.
+         *
+         * @param[in]  aFrame     The transmitted frame.
+         * @param[in]  aError     OT_ERROR_NONE when the frame was transmitted successfully,
+         *                        OT_ERROR_NO_ACK when the frame was transmitted but no ACK was received,
+         *                        OT_ERROR_CHANNEL_ACCESS_FAILURE tx failed due to activity on the channel,
+         *                        OT_ERROR_ABORT when transmission was aborted for other reasons.
+         * @param[in]  aChild     The child to which the frame was transmitted.
+         *
+         */
+        void HandleSentFrameToChild(const Mac::Frame &aFrame, otError aError, Child &aChild);
+
+        /**
+         * This callback method notifies that a pending `FrameChange` request (from `RequestFrameChange()`) is
+         * now processed and the indirect message for the child can be updated.
+         *
+         * @param[in]  aChild     The child to update.
+         */
+        void UpdateIndirectMessage(Child &aChild);
+
+        /**
+         * This callback method notifies that a requested frame change from `RequestFrameChange()` is processed.
+         *
+         * This callback indicates to the next layer that the indirect frame/message for the child can be safely
+         * updated.
+         *
+         * @param[in]  aChild     The child to update.
+         *
+         */
+        void HandleFrameChangeDone(Child &aChild);
+    };
+
+    /**
+     * This constructor initializes the data poll handler object.
+     *
+     * @param[in]  aInstance   A reference to the OpenThread instance.
+     *
+     */
+    explicit DataPollHandler(Instance &aInstance);
+
+    /**
+     * This method disables data poll handler and clears any state/info saved per child for indirect frame tx.
```

Comment:
What does "disables data poll handler" mean?

Reply:
I changed the comment to mention it would only clear data (which is what it does).  Thanks